integration/hive/src/main/java/org/apache/carbondata/hive/CarbonHiveRecordReader.java

```diff
@@ -111,58 +108,46 @@ private void initialize(InputSplit inputSplit, Configuration conf) throws IOExce
     } else {
       columnTypes = TypeInfoUtils.getTypeInfosFromTypeString(columnTypeProperty);
     }
+
+    if (valueObj == null) {
+      valueObj = new ArrayWritable(Writable.class, new Writable[columnTypes.size()]);
+    }
```

Comment:
Actually the data structure should be consistent, initially we were just returning the project columns and the Arraywritable was having variable length but in Parquet and ORC both implementation the ArrayWritable length was equivalent to the number of columns in table . It was causing issues in TPCh queries so that's why made changes to have Arraywritable size equivalent to number of columns and then populate the data at respective position